v2/chunk.go

```diff
@@ -0,0 +1,21 @@
+package pie
+
+import "fmt"
+
+// Chunk Split slice to chunks
+func Chunk[T any](ss []T, chunkSize int, callback func(ss []T) (stopped bool)) {
```

Comment:
This function feels like it might be ultra specific to your use case. Do you have some examples you can provide?